### ISSUE TITLE
UI: `e.templateRef is undefined` when using inline template

### ISSUE THREAD

**Issue by antoinetran** [argoproj/argo-workflows#9167]
## Checklist

 

* [ ] Double-checked my configuration.
* [ ] Tested using the latest version.
* [ ] Used the Emissary executor.

## Summary

What happened/what you expected to happen?
Creates a simple workflow with inline template. It should succeed.

What version are you running?
Argo v3.2.11

## Diagnostics

Paste the smallest workflow that reproduces the bug. We must be able to run the workflow.

From https://raw.githubusercontent.com/argoproj/argo-workflows/master/examples/steps-inline-workflow.yaml:
```yaml
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: steps-inline-
  labels:
    workflows.argoproj.io/test: "true"
  annotations:
    workflows.argoproj.io/description: |
      This workflow demonstrates running a steps with inline templates.
    workflows.argoproj.io/version: ">= 3.2.0"
spec:
  entrypoint: main
  templates:
    - name: main
      steps:
        - - name: a
            inline:
              container:
                image: argoproj/argosay:v2
```

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by antoinetran** [comment]
This bug is exactly like https://github.com/argoproj/argo-workflows/issues/7749

**Comment by antoinetran** [comment]
Error image:
![image](https://user-images.githubusercontent.com/14093758/179522922-2fbdaf73-6f8b-4d86-a069-a034bb059aa9.png)

**Comment by terrytangyuan** [comment]
Thanks for reporting. This should be fixed in https://github.com/argoproj/argo-workflows/pull/9168.

**Comment by agilgur5** [comment]
Marking this as duplicate of #7749 per https://github.com/argoproj/argo-workflows/pull/9168#discussion_r1574264448. #7605 did fix this, it was just never backported to 3.2.x
